sycl/source/detail/queue_impl.cpp

```diff
@@ -134,7 +134,11 @@ event queue_impl::memset(const std::shared_ptr<detail::queue_impl> &Self,
     if (MContext->is_host())
       return MDiscardEvents ? createDiscardedEvent() : event();
 
-    if (isInOrder()) {
+    // When a queue is recorded by a graph, the dependencies are managed in the
+    // graph implementaton. Additionally, CG recorded for a graph are outside of
+    // the in-order queue execution sequence. Therefore, these CG must not
+    // update MLastEvent.
+    if (isInOrder() && (getCommandGraph() == nullptr)) {
```

Comment:
Exactly. In-order queue keeps a track of the last command/event submitted to the Queue. When an in-order queue is recorded, the recorded events should not have impact on nominal tracking since these events are not executed directly.
So, if we submit A then record B and then submit C on the same in-order queue, C must wait for the completion of A and not B. That's why we've added this condition.